openadapt/db/crud.py

```diff
@@ -357,18 +359,29 @@ def _get(
     session: SaSession,
     table: BaseModelType,
     recording_id: int,
+    eager: bool = False,
 ) -> list[BaseModelType]:
     """Retrieve records from the database table based on the recording timestamp.
 
     Args:
         session (sa.orm.Session): The database session.
         table (BaseModel): The database table to query.
         recording_id (int): The recording id.
+        eager (bool): if true, implement eagerloading.
 
     Returns:
         list[BaseModel]: A list of records retrieved from the database table,
           ordered by timestamp.
     """
+    if eager:
+        return (
+            session.query(table)
+            .filter(table.recording_id == recording_id)
+            .order_by(table.timestamp)
+            .options(joinedload(table.recording))
+            .all()
+        )
+
```

Comment:
How about:
```
query = (
    session.query(table)
    .filter(table.recording_id == recording_id)
    .order_by(table.timestamp)
)

if eager:
    query = query.options(joinedload(table.recording))

return query.all()
```

Reply:
It looks like this functionality won't accomplish the same as in https://github.com/OpenAdaptAI/OpenAdapt/pull/822/commits/4f68487e00b81354a44b261093f20feb88984bbf, since here it's always joining on `table.recording`, whereas previously it was:

- get_action_events:
```
.options(joinedload(ActionEvent.screenshot))
```

- get_screenshots:
```
.options(joinedload(Screenshot.action_event))
```

 - get_window_events:
```
.options(joinedload(WindowEvent.action_events))
```

If we want to reproduce this behavior I think we need to pass in additional parameters.

But before we do that, @Animesh404 can you please confirm you were able to reproduce the initial exception, and confirm that this fixes it?

Reply:
I tried both the methods and it was working alright, even with joining only table.recording in all functions